src/etl/permits_extractor.py

```
@@ -0,0 +1,953 @@
+                if self.state == "half-open":
+                    self.state = "closed"
+                    self.failure_count = 0
+                    logger.info("Circuit breaker closed after successful call")
+                return result
+            except Exception as e:
+                self.failure_count += 1
+                self.last_failure_time = datetime.now()
+                
+                if self.failure_count >= self.failure_threshold:
+                    self.state = "open"
+                    logger.error(f"Circuit breaker opened after {self.failure_count} failures")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)